test-tsd/tables.test-d.ts

```diff
@@ -65,4 +65,17 @@ const main = async () => {
   expectType<Pick<User, 'id'> | undefined>(
     await knexInstance.first('id').from('users_composite')
   );
+
+  //These tests simply check if type work by showing that it does not throw syntax error
+
+  knexInstance.schema.createTable('testTable',(table) => {
+    table.foreign('fkey_three').references('non_exist.id').withKeyName('non_for1').deferrable('deferred');
+    table.foreign('fkey_threee').references('non_exist.id').deferrable('deferred').withKeyName('non_for2');
+    table.integer('num').references('non_exist.id').deferrable('immediate').withKeyName('non_for3');
+    table.integer('num').references('non_exist.id').withKeyName('non_for4').deferrable('deferred').onDelete('CASCADE');
+    table.integer('num').references('non_exist.id').withKeyName('non_for5').deferrable('deferred').onDelete('CASCADE');
+    table.integer('num').references('id').inTable('non_exist').withKeyName('non_for6').deferrable('deferred').onDelete('CASCADE');
+    table.integer('num').references('id').withKeyName('non_for7').deferrable('deferred').inTable('non_exist').onDelete('CASCADE');
+  })
```

Comment:
@kibertoad test added. Should this test be here or should we add a different test file for this type of test?

Reply:
here is good!